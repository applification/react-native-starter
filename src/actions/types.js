export const CHANGE_CONNECTION_STATUS = 'change_connection_status';

export const FETCH_SUPPLIERS_REQUEST = 'fetch_supplier_request';
export const FETCH_SUPPLIERS_SUCCESS = 'fetch_suppliers_success';
export const FETCH_SUPPLIERS_FAILURE = 'fetch_suppliers_failure';
