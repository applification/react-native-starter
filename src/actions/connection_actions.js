import { CHANGE_CONNECTION_STATUS } from './types';

export const connectionState = ({ status }) => ({
  type: CHANGE_CONNECTION_STATUS,
  isConnected: status
});
