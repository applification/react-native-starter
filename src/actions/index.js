export * from './connection_actions';
export * from './supplier_actions';
