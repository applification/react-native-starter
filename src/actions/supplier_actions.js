import fetch from 'fetch-everywhere';
import {
  FETCH_SUPPLIERS_REQUEST,
  FETCH_SUPPLIERS_SUCCESS,
  FETCH_SUPPLIERS_FAILURE
} from '../actions/types';

import config from '../../config';

const ROOT_URL = config.RANDOM_USER_API;

export const fetchSuppliers = () => dispatch => {
  dispatch(fetchSuppliersRequest());
  return fetch(ROOT_URL)
    .then(res => res.json())
    .then(json => dispatch(fetchSuppliersSuccess(json.results)))
    .catch(ex => dispatch(fetchSuppliersFailure(ex)));
};

export function fetchSuppliersRequest() {
  const data = {
    loading: true
  };

  return {
    type: FETCH_SUPPLIERS_REQUEST,
    payload: data
  };
}

export function fetchSuppliersFailure(error) {
  const data = {
    loading: false,
    error
  };

  return {
    type: FETCH_SUPPLIERS_FAILURE,
    payload: data
  };
}

export function fetchSuppliersSuccess(suppliers) {
  const data = {
    loading: false,
    data: suppliers
  };

  return {
    type: FETCH_SUPPLIERS_SUCCESS,
    payload: data
  };
}
