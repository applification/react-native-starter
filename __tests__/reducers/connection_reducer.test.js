import connectionReducer, {
  initialState
} from '../../src/reducers/connection_reducer';
import * as actions from '../../src/actions/types';

it('should return an initial state', () => {
  expect(connectionReducer(initialState, { type: '_NULL' })).toMatchSnapshot();
});

it('handles action of type "CHANGE_CONNECTION_STATUS" offline', () => {
  const action = { type: actions.CHANGE_CONNECTION_STATUS, isConnected: false };
  expect(connectionReducer(initialState, action)).toMatchSnapshot();
});

it('handles action of type "CHANGE_CONNECTION_STATUS" online', () => {
  const action = { type: actions.CHANGE_CONNECTION_STATUS, isConnected: true };
  expect(connectionReducer(initialState, action)).toMatchSnapshot();
});
