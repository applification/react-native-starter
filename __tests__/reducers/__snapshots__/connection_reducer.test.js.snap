// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles action of type "CHANGE_CONNECTION_STATUS" offline 1`] = `
Object {
  "isConnected": false,
}
`;

exports[`handles action of type "CHANGE_CONNECTION_STATUS" online 1`] = `
Object {
  "isConnected": true,
}
`;

exports[`should return an initial state 1`] = `
Object {
  "isConnected": "unknown",
}
`;
